src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -371,5 +418,50 @@ public static object MarkAsNonPII(object arg)
         {
             return new NonPII(arg);
         }
+
+        /// <summary>
+        /// Creates a <see cref="LogEntry"/> by using the provided event level, exception argument, string argument and arguments list.
+        /// </summary>
+        /// <param name="level"><see cref="EventLevel"/></param>
+        /// <param name="innerException"><see cref="Exception"/></param>
+        /// <param name="message">The log message.</param>
+        /// <param name="args">An object array that contains zero or more objects to format.</param>
+        private static LogEntry WriteEntry(EventLevel level, Exception innerException, string message, params object[] args)
+        {
+            if (string.IsNullOrEmpty(message))
+                return null;
+
+            if (innerException != null)
+            {
+                // if PII is turned off and 'innerException' is a System exception only display the exception type
+                if (!IdentityModelEventSource.ShowPII && !LogHelper.IsCustomException(innerException))
+                    message = string.Format(CultureInfo.InvariantCulture, "Message: {0}, InnerException: {1}", message, innerException.GetType());
+                else // otherwise it's safe to display the entire exception message
```

Comment:
nit: I would modify the comments...if PII is turned on, then display the entire exception message. 
by default, we only log the exception type.

Reply:
innerException could be a System or IdentityModel exception.
When PII is turned OFF only the type of System exception is displayed. However, it is safe to display an IdentityModel exception in clear text.